src/Raven.Server/SqlMigration/Model/MigrationSettings.cs

```diff
@@ -2,10 +2,18 @@
 
 namespace Raven.Server.SqlMigration.Model
 {
+    public enum EmbeddedDocumentSqlKeysStorage
+    {
+        None,
+        OnDocument,
+        OnMetadata,
+    }
+    
     public class MigrationSettings
     {
         public List<RootCollection> Collections { get; set; }
         public int BatchSize { get; set; } = 1000;
         public int? MaxRowsPerTable { get; set; }
+        public Dictionary<string, EmbeddedDocumentSqlKeysStorage> CollectionsEmbeddedReferencesSqlKeysConfigurations = new();
```

Comment:
I think that instead of having it as a separate option it should be rather a property of imported table. Maybe a property of `EmbeddedCollection`? 

Do we want to expose such option not only for embedded objects? Let's talk directly about that

Reply:
I've prepared a commit with `EmbeddedDocumentSqlKeysStorage` as a property of `EmbeddedCollection`, and also `ReferenceInformation`.